src/dashboard/home-dashboard.test.tsx

```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import { render, fireEvent } from "@testing-library/react";
+import HomeDashboard from "./home-dashboard.component";
+import { Router } from "react-router-dom";
+import { createMemoryHistory } from "history";
+
+const match = { params: { id: 1 }, isExact: true, path: "", url: "" };
+function renderWithRouter(
+  ui,
+  {
+    route = "/",
+    history = createMemoryHistory({ initialEntries: [route] })
+  } = {}
+) {
+  return {
+    ...render(<Router history={history}>{ui}</Router>),
+    history
+  };
+}
+it("renders without failing", () => {
+  renderWithRouter(<HomeDashboard match={match} />);
+});
+
+it("directs to patient search on click", () => {
+  const wrapper = renderWithRouter(<HomeDashboard match={match} />);
+  const patientSearchLink = wrapper.getByTestId("patientSearchLink");
```

Comment:
`getByTestId` should only be used as a last resort, since it departs from `@testing-library/react`'s mission of only interacting with the UI in the same way that users do. See https://testing-library.com/docs/dom-testing-library/api-queries#bytestid which explains this.

Instead, I think you can do `wrapper.getByText('Search for patient')` and then fire the click on that instead. DOM events bubble, so the `<Link>` will also handle the click event and trigger the reroute.

Reply:
aah thanks for that, I didn't quite figure out a way to get the Link.